internal/api/v1/task.go

```diff
@@ -7,85 +7,103 @@ import (
 	ierr "github.com/flexprice/flexprice/internal/errors"
 	"github.com/flexprice/flexprice/internal/logger"
 	"github.com/flexprice/flexprice/internal/service"
+	"github.com/flexprice/flexprice/internal/temporal/models"
+	temporalservice "github.com/flexprice/flexprice/internal/temporal/service"
 	"github.com/flexprice/flexprice/internal/types"
 	"github.com/gin-gonic/gin"
+	"github.com/samber/lo"
 )
 
 type TaskHandler struct {
-	taskService service.TaskService
-	logger      *logger.Logger
+	service         service.TaskService
+	temporalService temporalservice.TemporalService
+	log             *logger.Logger
 }
 
-func NewTaskHandler(taskService service.TaskService, logger *logger.Logger) *TaskHandler {
+func NewTaskHandler(
+	service service.TaskService,
+	temporalService temporalservice.TemporalService,
+	log *logger.Logger,
+) *TaskHandler {
 	return &TaskHandler{
-		taskService: taskService,
-		logger:      logger,
+		service:         service,
+		temporalService: temporalService,
+		log:             log,
 	}
 }
 
-// CreateTask godoc
 // @Summary Create a new task
-// @Description Create a new import/export task
+// @Description Create a new task for processing files asynchronously
 // @Tags Tasks
 // @Accept json
 // @Produce json
 // @Security ApiKeyAuth
-// @Param task body dto.CreateTaskRequest true "Task details"
-// @Success 201 {object} dto.TaskResponse
+// @Param task body dto.CreateTaskRequest true "Task configuration"
+// @Success 202 {object} dto.TaskResponse
 // @Failure 400 {object} ierr.ErrorResponse
 // @Failure 500 {object} ierr.ErrorResponse
 // @Router /tasks [post]
 func (h *TaskHandler) CreateTask(c *gin.Context) {
 	var req dto.CreateTaskRequest
 	if err := c.ShouldBindJSON(&req); err != nil {
-		h.logger.Error("failed to bind request", "error", err)
 		c.Error(ierr.WithError(err).
-			WithHint("Please check the request payload").
+			WithHint("Invalid request format").
 			Mark(ierr.ErrValidation))
 		return
 	}
 
-	task, err := h.taskService.CreateTask(c.Request.Context(), req)
+	resp, err := h.service.CreateTask(c.Request.Context(), req)
 	if err != nil {
-		h.logger.Error("failed to create task", "error", err)
 		c.Error(err)
 		return
 	}
 
-	c.JSON(http.StatusCreated, task)
+	// Start the temporal workflow for async processing using the unified method
+	workflowRun, err := h.temporalService.ExecuteWorkflow(c.Request.Context(), types.TemporalTaskProcessingWorkflow, resp.ID)
+
+	if err != nil {
+		h.log.Error("failed to start temporal workflow", "error", err, "task_id", resp.ID)
+		c.Error(err)
+		return
+	}
+
+	c.JSON(http.StatusOK, models.TemporalWorkflowResult{
+		Message:    "task processing workflow started successfully",
+		WorkflowID: workflowRun.GetID(),
+		RunID:      workflowRun.GetRunID(),
```

Comment:
Ideally workflow and run id should be stored in task metadata or i think even better to create a column.

Also need to implement a retry api as well so that we can ask temporal to retry the task and update the run id after retry

Also need to check if we can cancel a task? I think temporal allows terminating a task so maybe we can use that as well